src/blocks/vpn.rs

```diff
@@ -0,0 +1,270 @@
+//! properly the binary should be executable without root privileges.
+//!
+//! # Example
+//!
+//! Shows the current vpn network state:
+//!
+//! ```toml
+//! [[block]]
+//! block = "vpn"
+//! driver = "nordvpn"
+//! interval = 10
+//! format_connected = "VPN: $icon "
+//! format_disconnected = "VPN: $icon "
+//! state_connected = "good"
+//! state_disconnected = "warning"
+//! ```
+//!
+//! Possible values for `state_connected` and `state_disconnected`:
+//!
+//! ```
+//! Warning
+//! Critical
+//! Good
+//! Info
+//! Idle
+//! ```
+//! `Option::None` defaults to `idle`.
+//!
+//! # Icons Used
+//!
+//! - `net_vpn`
+//! - `net_wired`
+//! - `net_down`
+//! - country code flags (if supported by font)
+//!
+//! Flags: They are not icons but unicode glyphs. You will need a font that
+//! includes them. Tested with: <https://www.babelstone.co.uk/Fonts/Flags.html>
+
+use regex::Regex;
+use std::process::Stdio;
+use tokio::process::Command;
+
+use crate::util::country_flag_from_iso_code;
+
+use super::prelude::*;
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(rename_all = "snake_case")]
+enum DriverType {
+    #[default]
+    Nordvpn,
+}
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(default)]
+pub struct Config {
+    driver: DriverType,
+    #[default(10.into())]
+    interval: Seconds,
+    format_connected: FormatConfig,
+    format_disconnected: FormatConfig,
+    state_connected: State,
+    state_disconnected: State,
+}
+
+enum Status {
+    Connected {
+        country: String,
+        country_flag: String,
+    },
+    Disconnected,
+    Error,
+}
+
+impl Status {
+    fn icon(&self) -> &str {
+        match self {
+            Status::Connected { .. } => "net_vpn",
+            Status::Disconnected => "net_wired",
+            Status::Error => "net_down",
+        }
+    }
+
+    fn flag(&self) -> &str {
+        match self {
+            Status::Connected { country_flag, .. } => country_flag.as_str(),
+            _ => "",
+        }
+    }
+
+    fn country(&self) -> &str {
+        match self {
+            Status::Connected { country, .. } => country.as_str(),
+            _ => "",
+        }
+    }
+}
+
+pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
+    api.set_default_actions(&[(MouseButton::Left, None, "toggle")])
+        .await?;
+
+    let mut widget = Widget::new();
+
+    let format_connected = config.format_connected.with_default(" VPN: $icon ")?;
+    let format_disconnected = config.format_disconnected.with_default(" VPN: $icon ")?;
+
+    let driver: Box<dyn Driver> = match config.driver {
+        DriverType::Nordvpn => Box::new(NordVpnDriver::new().await),
+    };
+
+    loop {
+        let status = driver.get_status().await?;
+
+        widget.set_values(map!(
+            "icon" => Value::icon(api.get_icon(status.icon())?),
+            "country" => Value::text(status.country().to_string()),
+            "flag" => Value::text(status.flag().to_string()),
+
+        ));
```

Comment:
I think it's better to only add `country` and `flag` if they exist, i.e. if `status` is `Connected`.